### DISCUSSION TITLE
Geräte Firmware HmIP-FSM16 - 1.18.20 wird nicht installiert

### DISCUSSION THREAD

**Discussion by Tedon75** [OpenCCU/OpenCCU#1233] [🇩🇪 Fragen&Antworten]
Guten Tag,

meine eingestzte CCU3, mit der SW RaspberryMatic 3.57.4.20210320 zeigt für das Gerät HmIP-FSM16 ein Geräteupdate 1.18.20  an.
Das Update-File wurde heruntergeladen und in der RaspberryMatic importiert. Nach ca. 3 Wochen wurde die Geräte immer noch nicht aktualisiert.
Wie kann überprüft werden warum das Geräteupdate nicht installiert wird bzw. gibt es eine Beschreibung um das Update manuell auszulösen?
Anmerkung: - gleiches Verhalten auch bei der  RaspberryMatic 3.57.4.20210320 Vorgängerversion

Vielen Dank für Eure Unterstützung.

**Comment by Baxxy13** [comment]
> Nach ca. 3 Wochen wurde die Geräte immer noch nicht aktualisiert.

Das heißt du hängst in folgendem Status fest und der Update-Button erscheint an dieser Stelle nicht?
[Beispiel HmIP-PSM "Downgrade"]
![PSM_Firmware_Update](https://user-images.githubusercontent.com/54898249/115230737-08dcb800-a115-11eb-8e9b-e5576b587f3e.JPG)

Der ganze Geräte-Firmware-Update-Mechanismus steckt meines Wissens im ClosedSource Teil (HmIP-Server ?) der eQ-3 Software.
RaspberryMatic nutzt die gleichen Routinen und eine Möglichkeit irgendwie Einfluss auf den Prozess zu nehmen gibt es nicht.

  
> Wie kann überprüft werden warum das Geräteupdate nicht installiert wird

Man könnte sich die Logdatei ( _/var/log/hmserver.log_ ) anschauen. Dort nach Einträgen suchen die...

- "OTAU Handler"
- "OTAU RF Handler"
- "suspended"
- "aborted"

 
enthalten.


> gibt es eine Beschreibung um das Update manuell auszulösen?

Nein, da gibt es keine Beschreibung. Es gibt nur diesen Mechanismus.

- Update auf die Zentrale laden
- warten bis der Update Button erscheint und diesen dann drücken

Tipps aus dem Homematic-Forum:
(ohne Gewähr)

- immer nur eine Aktor / Sensorart gleichzeitig updaten
- einen integrierten HmIP-HAP während der Updatephase aus dem System entfernen
- den eigenen "Grund-Duty-Cycle" möglichst gering halten
- auf keinen Fall die Zentrale neu starten oder den Aktor / Sensor stromlos machen
- den Aktor / Sensor in die Nähe der Zentrale bringen (1-2m)

**Comment by Tedon75** [comment]
ja korrekt, der update Button erscheint nicht für die 16A Akktuatoren (HmIP-FSM16)

Note: die bisherigen updates der Komponenten haben funktioniert

**Reply by jens-maus** [reply]
Meist hilft es einen Blick in das `/var/log/hmserver.log` zu werfen. Dort sollte der Fortschritt der OTAU Übertragungen irgendwie erkennbar sein.

**Comment by Tedon75** [comment]
Im Anhang habe ich einen Auszug vom Log-Fil hmserver.log angehangen.  Die Updates scheinen auf ausgesetzt zu stehen "suspended for device(s)"

[homematic-raspi-2021-04-20.log](https://github.com/jens-maus/RaspberryMatic/files/6343050/homematic-raspi-2021-04-20.log)

**Reply by jens-maus** [reply]
> Im Anhang habe ich einen Auszug vom Log-Fil hmserver.log angehangen. Die Updates scheinen auf ausgesetzt zu stehen "suspended for device(s)"

Na wenn die "suspended" werden, dann liegt das meistens daran das der Grund DutyCycle zu hoch ist, denn ab einem DutyCycle von ca. 60% stellt der HMIPServer die Auslieferung der Updates komplett ein um die Grundfunktionalität nicht zu gefährden. Ergo müsste man also erst einmal schauen wie hoch der DutyCycle der Zentrale so über einen längeren Zeitraum ist.

**Comment by Baxxy13** [comment]
Im Auszug sind nur 2 relevante Zeilen zu finden...
`Apr 20 11:14:48 de.eq3.cbcs.server.core.otau.DeviceBackgroundUpdateSubsystem INFO  [vert.x-eventloop-thread-7] AP 3014F711A0001F5A4993FA2F: OTAU Handler 32077286-5d7d-4f47-bf4d-e0c31f4ba5dc continues for device(s) [3014F711A0000B9BE994656B]`
und
`Apr 20 11:35:48 de.eq3.cbcs.server.core.otau.DeviceBackgroundUpdateSubsystem INFO  [vert.x-eventloop-thread-7] AP 3014F711A0001F5A4993FA2F: OTAU RF Handler 32077286-5d7d-4f47-bf4d-e0c31f4ba5dc aborted for device(s) [3014F711A0000B9BE994656B]`

Das hier...
`Apr 20 11:14:48 de.eq3.cbcs.server.core.otau.DeviceBackgroundUpdateSubsystem INFO  [vert.x-eventloop-thread-7] SYSTEM: There are 1 APs queued with updatable devices (RF)`
...deutet auf einen aktiven AP im System hin.

Also bitte mal alle Fakten auf den Tisch legen sonst wird das hier nichts!

**Comment by Tedon75** [comment]
Vielen Dank für Ihre Rückmeldung.
Ergänzend möchte ich erwähnen das ich vor ca. 1 Jahr von Homematic AP -Variante auf die CCu Variante umgestiegen bin. D.h. sollte ich ggf. "Anfängerfragen stellen" entschuldigen Sie das bitte. 

Ich betreibe in meinem System keinen Access Point (AP), aus diesem Grund verwundert mich die Fehlerdokumentation.

Das System zeigt einen Duty Cycle im Bereich von 40% bis 60%. Lösche ich die Aktuatoren HmIP-FSM16 aus der Geräteverwaltung zeigt die erste Messung einen Duty Cycle von 3% - 6%.

Können die Geräte noch in der Cloud der Homematic AP Variante hängen obwohl diese auf den Werkzustand zurückgesetzt wurden?
Sobald ich das Gerät wieder anmelde erscheint wieder die AP Info Mitteilung im Log-File:

de.eq3.cbcs.server.core.otau.DeviceBackgroundUpdateSubsystem INFO [vert.x-eventloop-thread-7] AP 3014F711A0001F5A4993FA2F: OTAU Handler 32077286-5d7d-4f47-bf4d-e0c31f4ba5dc continues for device(s) [3014F711A0000B9BE994656B]

Vielen Dank für Ihre Unterstützung

**Reply by Baxxy13** [reply]
> Ich betreibe in meinem System keinen Access Point (AP), aus diesem Grund verwundert mich die Fehlerdokumentation.

Das war eine falsche Interpretation meinerseits. Nach etwas Recherche wurde mir klar das die Zentrale selbst der "AP" ist. Dahingehend also alles gut.


> zeigt einen Duty Cycle im Bereich von 40% bis 60%.

Das ist während der Übertragung der IP-Gerätefirmware normal. Hier ein visualisiertes Beispiel.
![DC_IP_Updates](https://user-images.githubusercontent.com/54898249/115955090-e40d8980-a4f4-11eb-81e5-c580441b1595.JPG)

Letztlich hilft wohl nur Geduld und wie schon geschrieben, keinesfalls den Aktor stromlos machen oder die Zentrale neu starten.
